anomdetect/src/test/java/com/expedia/adaptivealerting/anomdetect/detect/outlier/algo/forecasting/LegacySeasonalNaiveFactoryTest.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2018-2019 Expedia Group, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.expedia.adaptivealerting.anomdetect.detect.outlier.algo.forecasting;
+
+import com.expedia.adaptivealerting.anomdetect.detect.AbstractDetectorFactoryTest;
+import com.expedia.adaptivealerting.anomdetect.detect.AnomalyType;
+import com.expedia.adaptivealerting.anomdetect.forecast.interval.algo.expwelford.ExponentialWelfordIntervalForecaster;
+import com.expedia.adaptivealerting.anomdetect.forecast.point.algo.seasonalnaive.SeasonalNaivePointForecaster;
+import lombok.val;
+import org.junit.Test;
+
+import static com.expedia.adaptivealerting.anomdetect.forecast.point.algo.seasonalnaive.SeasonalNaivePointForecasterParams.DEFAULT_MISSING_VALUE_PLACEHOLDER;
+import static org.junit.Assert.assertEquals;
+
+@Deprecated // Use ForecastingDetectorFactory
+public class LegacySeasonalNaiveFactoryTest extends AbstractDetectorFactoryTest {
+    private static final double TOLERANCE = 0.001;
+    private static final int ONE_WEEK_IN_SECONDS = 60 * 24 * 7;
+    private static final int ONE_MINUTE_IN_SECONDS = 60;
+
+    @Test
+    public void testBuildDetector() {
+        val factoryUnderTest = new LegacySeasonalNaiveDetectorFactoryProvider();
+        val document = readDocument("seasonal-naive");
+        val detector = factoryUnderTest.buildDetector(document);
+        val seasonalNaive = (SeasonalNaivePointForecaster) detector.getPointForecaster();
+        val seasonalNaiveParams = seasonalNaive.getParams();
+        val welford = (ExponentialWelfordIntervalForecaster) detector.getIntervalForecaster();
```

Comment:
Welford won't work for seasonal series. You suggested a modified version of the Welford for Holt-Winters, but that modification won't work for the seasonal naive case because there's only a single cycle (so variance is always 0).

My suggestion for the short term is to use a multiplicative interval forecaster. Longer term we will need some other kind of interval forecaster here -- maybe we can modify the power law interval forecaster to work in an online fashion (that should be easy enough to do). Another possibility is to add a new interval forecaster that estimates variance based on a local window around the current timestamp, and uses that to generate a threshold.